dbt_platform_helper/utils/validation.py

```diff
@@ -553,8 +559,91 @@ def validate_platform_config(config, disable_aws_validation=False):
     _validate_environment_pipelines_triggers(enriched_config)
     _validate_codebase_pipelines(enriched_config)
     validate_database_copy_section(enriched_config)
+
     if not disable_aws_validation:
         _validate_s3_bucket_uniqueness(enriched_config)
+        _validate_redis_versions(config)
+        _validate_opensearch_versions(config)
+
+
+def _validate_redis_versions(config):
+
+    supported_redis_versions = []
+
+    if cache_refresh_required("redis"):
+        supported_redis_versions = get_redis_supported_versions()
+    else:
+        supported_redis_versions = read_supported_versions_from_cache("redis")
+
+    redis_extensions = []
+    redis_environments_with_failure = []
+
+    for extension in config.get("extensions").keys():
+
+        if config.get("extensions").get(extension).get("type") == "redis":
+            redis_extensions.append(config["extensions"][extension])
+
+    for redis_extension in redis_extensions:
+        environments = redis_extension.get("environments", {})
+        if isinstance(environments, dict):
+            for environment in environments.keys():
+                env_config = environments.get(environment, {})
+                if isinstance(env_config, dict):
+                    redis_engine_version = env_config.get("engine")
+                    if redis_engine_version not in supported_redis_versions:
+                        redis_environments_with_failure.append(
+                            [extension, environment, redis_engine_version]
```

Comment:
This is 5 levels of nesting. Could be useful to break into methods to make it clear what each level is trying to do, and help turn some of the if statements into guard clauses.